openff/interchange/interop/gromacs/export/_export.py

```diff
@@ -222,7 +222,38 @@ def _write_atoms(
         top.write("[ atoms ]\n")
         top.write(";index, atom type, resnum, resname, name, cgnr, charge, mass\n")
 
-        for atom in molecule_type.atoms:
+        def _adjust_charges(charges: numpy.array, tolerance=8) -> numpy.array:
+            """
+            Adjust charges so that written charge for a molecule type is 0.
+            """
+            rounded_charges = numpy.round(charges, tolerance)
+            total_charge = numpy.round(numpy.sum(charges_to_write), 0)
+
+            def _rounding_error(_arr, _sum, _tolerance):
+                return numpy.round(numpy.sum(_arr) - _sum, _tolerance)
```

Comment:
(not blocking) My external perspective here is that needing to come back to this definition repeatedly as this is called in multiple places decreases readability substantially, and that the line length is about the same if `numpy.round` were called directly on each line.